src/test/regress/sql/multi_sequence_default.sql

```diff
@@ -0,0 +1,215 @@
+SET citus.shard_replication_factor TO 1;
+SELECT start_metadata_sync_to_node('localhost', :worker_1_port);
+-- apply workaround
+SELECT run_command_on_workers('DROP SCHEMA sequence_default_1 CASCADE');
+-- now the sequence is gone from the worker as well
+\c - - - :worker_1_port
+INSERT INTO seq_test_5 VALUES (1, 2) RETURNING *;
+\c - - - :master_port
+SET citus.shard_replication_factor TO 1;
+SELECT start_metadata_sync_to_node('localhost', :worker_1_port);
+
+
+-- check some more complex cases
+CREATE SEQUENCE seq_6;
+CREATE TABLE seq_test_6 (x int, t timestamptz DEFAULT now(), s int DEFAULT nextval('seq_6'), m int) PARTITION BY RANGE (t);
+SELECT create_distributed_table('seq_test_6','x');
+-- shouldn't work since x is the partition column
+ALTER TABLE seq_test_6 ALTER COLUMN x SET DEFAULT nextval('seq_6');
+-- should work since both s and m have int type
+ALTER TABLE seq_test_6 ALTER COLUMN m SET DEFAULT nextval('seq_6');
+
+
+-- It is possible for a partition to have a different DEFAULT than its parent
+CREATE SEQUENCE seq_7;
+CREATE TABLE seq_test_7 (x text, s bigint DEFAULT nextval('seq_7'), t timestamptz DEFAULT now()) PARTITION BY RANGE (t);
+SELECT create_distributed_table('seq_test_7','x');
+CREATE SEQUENCE seq_7_par;
+CREATE TABLE seq_test_7_par (x text, s bigint DEFAULT nextval('seq_7_par'), t timestamptz DEFAULT now());
+ALTER TABLE seq_test_7 ATTACH PARTITION seq_test_7_par FOR VALUES FROM ('2021-05-31') TO ('2021-06-01');
+-- check that both sequences are in worker
+\c - - - :worker_1_port
+\d seq_7
+\d seq_7_par
+\c - - - :master_port
+SET citus.shard_replication_factor TO 1;
+SELECT start_metadata_sync_to_node('localhost', :worker_1_port);
+
+
+-- Check that various ALTER SEQUENCE commands
+-- are not allowed for a distributed sequence for now
+CREATE SEQUENCE seq_8;
+CREATE SCHEMA sequence_default_8;
+-- can change schema in a sequence not yet distributed
+ALTER SEQUENCE seq_8 SET SCHEMA sequence_default_8;
+ALTER SEQUENCE sequence_default_8.seq_8 SET SCHEMA public;
+CREATE TABLE seq_test_8 (x int, y int DEFAULT nextval('seq_8'));
+SELECT create_distributed_table('seq_test_8', 'x');
+-- cannot change sequence specifications
+ALTER SEQUENCE seq_8 AS bigint;
+ALTER SEQUENCE seq_8 INCREMENT BY 2;
+ALTER SEQUENCE seq_8 MINVALUE 5 MAXVALUE 5000;
+ALTER SEQUENCE seq_8 START WITH 6;
+ALTER SEQUENCE seq_8 RESTART WITH 6;
+ALTER SEQUENCE seq_8 NO CYCLE;
+ALTER SEQUENCE seq_8 OWNED BY seq_test_7;
+-- cannot change schema in a distributed sequence
+ALTER SEQUENCE seq_8 SET SCHEMA sequence_default_8;
+DROP SCHEMA sequence_default_8;
+
+
+-- cannot use more than one sequence in a column default
+CREATE SEQUENCE seq_9;
+CREATE SEQUENCE seq_10;
+CREATE TABLE seq_test_9 (x int, y int DEFAULT nextval('seq_9') - nextval('seq_10'));
+SELECT create_distributed_table('seq_test_9', 'x');
+
+
+-- clean up
+DROP TABLE seq_test_0, seq_test_1, seq_test_2, seq_test_3, seq_test_4, seq_test_5, seq_test_6, seq_test_7, seq_test_8, seq_test_9;
```

Comment:
Usually we create a schema for a test and drop the schema at the end so that:
- We don't need to drop the tables one by one
- After the test is done, it doesn't affect other tests since we drop the schema
- The test itself is isolated and doesn't depend on the other tests

Currently there are a lot of tests that depend on some other tests(which I believe we should get rid of)

Reply:
Addressed this in #5046